src/main/java/picard/sam/AddOATag.java

```diff
@@ -0,0 +1,131 @@
+            "<br />"+
+            "<h4>Usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar AddOATag \\<br />" +
+            "      L=some_picard.interval_list \\<br />" +
+            "      I=sorted.bam \\<br />" +
+            "      O=fixed.bam <br />"+
+            "</pre>";
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "SAM or BAM input file")
+    public String INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "SAM or BAM file to write merged result to")
+    public String OUTPUT;
+
+    @Argument(shortName = "L", doc = "If provided, only records that overlap given interval list will have the OA tag added.", optional = true)
+    public File INTERVAL_LIST;
+
+    /**
+     * Original Alignment tag key.
+     */
+    public static final String OA = "OA";
+    private static final Log log = Log.getInstance(AddOATag.class);
+
+    @Override
+    protected int doWork() {
+        try (final SamReader reader = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(IOUtil.getPath(INPUT));
+             final SAMFileWriter writer = new SAMFileWriterFactory().makeWriter(reader.getFileHeader(), true, IOUtil.getPath(OUTPUT), REFERENCE_SEQUENCE)) {
+                writer.setProgressLogger(
+                        new ProgressLogger(log, (int) 1e7, "Wrote", "records"));
+
+                final OverlapDetector overlapDetector = getOverlapDetectorFromIntervalListFile(INTERVAL_LIST, 0, 0);
+                for (final SAMRecord rec : reader) {
+                    if (overlapDetector == null || overlapDetector.overlapsAny(rec)) {
+                        setOATag(rec);
+                    }
+                    writer.addAlignment(rec);
+                }
+        } catch (IOException e) {
+            log.error(e);
+            return 1;
+        }
+
+        return 0;
+    }
+
+    // Take an interval list file and convert it to an overlap detector, can add left and right padding
+    static OverlapDetector<Interval> getOverlapDetectorFromIntervalListFile(File intervalList, int lhsBuffer, int rhsBuffer) {
+        if (intervalList == null) {
+            return null;
+        }
+        List<Interval> intervals = IntervalList.fromFile(intervalList).uniqued().getIntervals();
+        OverlapDetector<Interval> detector = new OverlapDetector<>(lhsBuffer, rhsBuffer);
+        detector.addAll(intervals, intervals);
+        return detector;
+    }
+
+    // format OA tag string according to the spec
+    //TODO: Move this to htsjdk once https://github.com/samtools/hts-specs/pull/193 is merged
+    private void setOATag(SAMRecord rec) {
+        if (rec.getReferenceName().contains(",")) {
+            throw new PicardException(String.format("Reference name for record %s contains a comma character.", rec.getReadName()));
+        }
+        final String oAValue;
```

Comment:
This is unreadable (to me)...if we are sticking to lower-case first letter, this shoudl be `oaValue`...sorry.